### DISCUSSION TITLE
Now that Script tag is out, should we update the integration of Google Analytics? [Next 11]

### DISCUSSION THREAD

**Discussion by GorvGoyl** [vercel/next.js#26210] [Help]
Official [Google Analytics integration example](https://github.com/vercel/next.js/blob/canary/examples/with-google-analytics/pages/_document.js) states that script should be placed in head section with `async` tag:

```
    <Head>
          {/* Global Site Tag (gtag.js) - Google Analytics */}
          <script
            async
            src={`https://www.googletagmanager.com/gtag/js?id=${GA_TRACKING_ID}`}
          />
          <script
            dangerouslySetInnerHTML={{
              __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${GA_TRACKING_ID}', {
              page_path: window.location.pathname,
            });
          `,
            }}
          />
        </Head>
```

How can [Next/Script](https://nextjs.org/blog/next-11#script-optimization) be used here?

**Comment by olaj** [comment]
I had some issues with this but it seems to work to just change it to the `<Script>` tag.

The issue I had was that I was loading the scripts in the `<Head>` tag in _document.tsx ... and that didn't work. when i moved the tag to _app.tsx and outside the `<Head>` tag it seems to work fine.

Maybe that should be clarified in the documentation?

**Comment by yokinist** [comment]
FYI:

> With `next/script`, you no longer need to wrap scripts in `next/head`. Further, `next/script` should **not** be used in `pages/_document.js` as `next/script` has client-side functionality to ensure loading order.

https://github.com/vercel/next.js/commit/d10f4e1607bee47dea0cf13d34deb0d2af5f7aff#diff-62dbdb4bc703f7997ea259ad68d8e823d9fafed1e9571155e01c4de21abb6b47R49

https://github.com/vercel/next.js/pull/26325 ~(※ not yet merged)~

~Perhaps the documentation will describe it soon.~ -> was merged

https://nextjs.org/docs/basic-features/script

**Reply by GorvGoyl** [reply]
In updated docs, google ana. example mentioned only one script to use but I think there are two scripts involved to make it work properly.

```
<script
            dangerouslySetInnerHTML={{
              __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${GA_TRACKING_ID}', {
              page_path: window.location.pathname,
            });
          `,
            }}
          />
```
